### PR TITLE
Add unit test for ScanPackageHandler

### PR DESCRIPTION
**Pull request by shochdoerfer**
As [discussed](https://github.com/repman-io/repman/pull/575), add unit test for the ScanPackageHandler.

### CODE REVIEW

File: tests/Unit/MessageHandler/Security/ScanPackageHandlerTest.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+     */
+    private PackageRepository $packageRepository;
+
+    protected function setUp(): void
+    {
+        $this->packageScanner = $this->createMock(PackageScanner::class);
+        $this->packageRepository = $this->createMock(PackageRepository::class);
+    }
+
+    public function testHandlerIgnoresNonExistentPackages(): void
```
Comment (marmichalski) [inline tests/Unit/MessageHandler/Security/ScanPackageHandlerTest.php:33]: This is covered here: https://github.com/repman-io/repman/blob/8b1d42ea876ee011b236791238a5d30420c83b2c/tests/Integration/MessageHandler/Security/ScanPackageHandlerTest.php#L40

**Review by marmichalski — COMMENTED** [review]
I believe it is fine how it is, but is there any specific reason you have decided to unit test this instead of adding more scenarios to already existing [integration test](https://github.com/repman-io/repman/blob/8b1d42ea876ee011b236791238a5d30420c83b2c/tests/Integration/MessageHandler/Security/ScanPackageHandlerTest.php)?

File: tests/Unit/MessageHandler/Security/ScanPackageHandlerTest.php (hunk 1 above)
Comment (shochdoerfer) [reply tests/Unit/MessageHandler/Security/ScanPackageHandlerTest.php:33]: The simple answer is: I did not expect this to be covered by an integration test ;) I checked the unit tests, found none, and added the test case. Surely, I can migrate them over the integration tests.

Comment (shochdoerfer) [reply tests/Unit/MessageHandler/Security/ScanPackageHandlerTest.php:33]: I checked the unit test and the integration test. I don't think it makes sense to merge them. The integration test currently "just" checks for errors (exceptions), the unit test is testing the flow in the handler. I don't think I could test the flow in an integration test (e.g. the code currently has 2 return statements. I don't think I could distinguish between both in the integration test).

**Review by akondas — APPROVED** [review]
We can merge this, thanks

**Pull request merged**